downloader/src/main/kotlin/VersionControlSystem.kt

```diff
@@ -387,7 +388,8 @@ abstract class VersionControlSystem(
         workingTree: WorkingTree,
         revision: String,
         path: String = "",
-        recursive: Boolean = false
+        recursive: Boolean = false,
+        options: Options = emptyMap() // VCS-specific options only
```

Comment:
Just adding an unused option to the function signature does not fix anything, so you should not say so in the commit message. Most importantly, these options are not read from the configuration file and passed on.

Reply:
I hope I have improved the commit message now, also emphasising that this options are only available if the VCS plugin is consumed using the API.

Reply:
> options are only available if the VCS plugin is consumed using the API.

I fear that's not enough. For the PR to be acceptable, VCS plugins should be configurable via `config.yml` entries, [just like package managers like *GradleInspector*](https://github.com/oss-review-toolkit/ort/blob/6bb98df0cc4071155d333a9296cec6d6ef52c4e6/model/src/main/resources/reference.yml#L107-L111), independently of any programmatic use / ORT Server context.

Reply:
I will make myself familiar with it.

Reply:
Now the VCS-specific configuration can also be modified via `config.yml`, I added an example in `reference.yml` (see [here](https://github.com/boschglobal/oss-review-toolkit/blob/a7c9533513dc8ea798af99f0bbe83da9a9eace46/model/src/main/resources/reference.yml#L170C1-L179C39)). 

It is also possible now to use the _ort download_ command on a problematic repository with a myriad of nested submodules and it was actually possible to reduce the submodule-activities to only the first layer of submodules.